app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -125,34 +85,8 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         else window.clearFlags(dontSleep)
     }
 
-    private fun updateOverlayIfVisible() {
-        if (browserOverlay?.isVisible == true) {
-            browserOverlay.updateFocusableViews()
-        }
-    }
-
-    private val onNavigationEvent = { event: NavigationEvent, value: String?,
-                                      autocompleteResult: InlineAutocompleteEditText.AutocompleteResult? ->
-        when (event) {
-            NavigationEvent.SETTINGS -> serviceLocator.screenController.showSettingsScreen(fragmentManager!!)
-            NavigationEvent.LOAD_URL -> {
-                (activity as MainActivity).onTextInputUrlEntered(value!!, autocompleteResult!!, UrlTextInputLocation.MENU)
-                setOverlayVisible(false)
-            }
-            NavigationEvent.LOAD_TILE -> {
-                (activity as MainActivity).onNonTextInputUrlEntered(value!!)
-                setOverlayVisible(false)
-            }
-            NavigationEvent.POCKET -> {
-                val (fragmentManager, activity) = Pair(fragmentManager, activity)
-                if (fragmentManager != null && activity != null) {
-                    serviceLocator.screenController.showPocketScreen(fragmentManager)
-                }
-            }
-            NavigationEvent.TURBO, NavigationEvent.PIN_ACTION, NavigationEvent.DESKTOP_MODE, NavigationEvent.BACK,
-            NavigationEvent.FORWARD, NavigationEvent.RELOAD -> { /* not handled by this object */ }
-        }
-        Unit
+    override fun onUrlChanged(session: Session, url: String) {
+        if (url == URLs.APP_URL_HOME) serviceLocator?.screenController?.showNavigationOverlay(fragmentManager, true)
```

Comment:
Should MainActivity own this? We can better decouple the webRenderFragment from the nav overlay this way.